### ISSUE TITLE
auto terminal scroll

### ISSUE THREAD

**Issue by andrewning** [JuliaEditorSupport/julia-vscode#486]
Is there a setting or a way I can make the terminal automatically scroll when I execute code or execute a block?  If I'm already at the bottom of the terminal then the new command automatically scrolls down.  However, when I see an error I scroll up through the terminal log to read it, then execute a line, and the terminal won't automatically scroll.  I find myself manually scrolling down to the bottom quite a bit (and because the way vscode executes sometimes there is a lot to scroll through).

**Comment by davidanthoff** [comment]
I don't think so... That whole part is controlled by VS Code, and I haven't seen an API for us to change things. I'll put this on the backlog, maybe at some future date this will be possible.

**Comment by cdsousa** [comment]
I also find the issue of having to manually scroll back to the bottom after reading some error output very annoying. There was a similar issue on the PowerShell plugin: https://github.com/PowerShell/vscode-powershell/issues/1257

**Comment by heliosdrm** [comment]
Maybe file an issue for VS Code to provide the API that would be needed for this? That would improve the usability of the REPL terminal a lot!

**Comment by cdsousa** [comment]
https://github.com/microsoft/vscode/issues/94208

**Comment by heliosdrm** [comment]
Thanks for the link to the VS Code issue. I hope it receives more support in the next days, to make it go into the backlog.

Probably other users of the VS Code extension for Julia also feel annoyed by having to scroll manually, and if they were aware of that issue might be happy to upvote it... but I`m not familiar with that workflow, and don't know if a call for support for that would improper behavior.

**Comment by cdsousa** [comment]
I don't know either. I thought I had linked the issue here before, but only after your comment I noticed that I had not. Now it is becoming a little rushed. Anyway, maybe you can open an issue requesting only the API.

**Comment by heliosdrm** [comment]
The following alleviates this issue - at least I avoid switching from keyboard to mouse in order to scroll down: use a keyboard shortcut to move focus to the terminal (normally ``ctrl + ` ``, but `ctrl + ñ` in my Spanish keyboard), and then press any key for the terminal to scroll down.

**Comment by cdsousa** [comment]
Wow, related vscode issue just received enough votes!

**Comment by Tyriar** [comment]
VS Code already has an API to scroll to the bottom as used in the called out PowerShell extension https://github.com/PowerShell/vscode-powershell/issues/1257#issuecomment-403232260. Can't Julia just use that after executing text?

**Comment by pfitzseb** [comment]
Yes, pretty sure we can.

**Comment by Tyriar** [comment]
That's definitely preferable to an upstream feature as it already has "smart locking" where you scroll up and it will stop scrolling automatically. The design intent is that any keyboard/user input (which may not be all input) will scroll to bottom.

**Comment by pfitzseb** [comment]
For sure.
It's kind of unfortunate that `vscode.Terminal` doesn't export a method for this, but the command + some checks for the active terminal should work just fine.
